### PR TITLE
Fix out of bounds in td-shim/src/lib.rs

### PR DESCRIPTION
**Pull request by liuw1**
Fix: #403

Signed-off-by: Wei Liu <wei3.liu@intel.com>

### CODE REVIEW

File: td-shim-tools/src/enroller.rs [hunk 1]
```diff
@@ -82,7 +83,10 @@ impl FirmwareRawFile {
         }
 
         // Update lengh field
```
Comment (jodh-intel) [inline td-shim-tools/src/enroller.rs:85]: Not part of this PR, but a typo: "length".

File: td-shim/src/lib.rs [hunk 2]
```diff
@@ -114,8 +114,8 @@ pub struct PayloadInfo {
 }
 
 /// Write three bytes from an integer value into the buffer.
-pub fn write_u24(data: u32, buf: &mut [u8]) {
-    assert!(data < 0xffffff);
+pub fn write_u24(data: u32, buf: &mut [u8; 3]) {
+    assert!(data <= 0xffffff);
```
Comment (jodh-intel) [inline td-shim/src/lib.rs:118]: Could you add a unit test for this change?

**Review by jodh-intel — COMMENTED** [review]
Thanks @liuw1 - a few comments...

File: td-shim-tools/src/enroller.rs (hunk 1 above)
Comment (liuw1) [reply td-shim-tools/src/enroller.rs]: Fixed

File: td-shim/src/lib.rs (hunk 2 above)
Comment (liuw1) [reply td-shim/src/lib.rs:118]: Sure, added.

**Review by jodh-intel — APPROVED** [review]
Thanks @liuw1.

lgtm
